src/SystemView.cpp

```diff
@@ -938,7 +939,14 @@ void SystemView::RefreshShips(void) {
 void SystemView::DrawShips(const double t, const vector3d &offset) {
 	m_shipLabels->Clear();
 	for(auto s = m_contacts.begin(); s != m_contacts.end(); s++) {
-		const vector3d pos = offset + (*s).second.OrbitalPosAtTime(t) * double(m_zoom);
+		vector3d rel_pos;
+		if ((*s).first->IsDocked() || (*s).first->IsLanded()) {
```

Comment:
I think that if you change this test too
'if ((*s).first->GetFlightState() != Ship::FlightState::FLYING) {'
Then it should catch all of the non-flying states including when docking is in progress. Hyperspace, etc.